README.md

```diff
@@ -6,35 +6,67 @@ A sidechain to Tezos by Marigold focused on higher throughput.
 
 ### Pre-requisites
 
-#### Esy Package Manager
+You'll need a one of the two package managers Deku is developed with:
 
-Marigold Sidechain is being developed with
-[esy](https://esy.sh/). `esy` can be installed with NPM or Yarn.
+#### With [Esy](https://esy.sh)
+Esy can be installed with NPM or Yarn.
 
 ```sh
 npm i -g esy # on Ubuntu you might need --unsafe-perms to work around EACCES issues
 # or
 yarn global add esy
 ```
 
-Alternatively, you can use the `nix` package manager with
-[flakes enabled](https://nixos.wiki/wiki/Flakes#Installing_flakes) to enter an
-environment preloaded with the correct dependencies, including `esy`:
+#### With Nix
 
+Nix can be installed with the following command on *nix systems:
+```shell
+sh <(curl -L https://nixos.org/nix/install) --no-daemon
 ```
-nix --experimental-features "nix-command flakes" run
+Additionally, you must enable the [flakes feature](https://nixos.wiki/wiki/Flakes):
+```
+echo 'experimental-features = nix-command flakes' >> /etc/nix/nix.conf
+```
+
+Nix automatically installs all the dependencies and devtools needed for development,
+including Docker, docker-compose, etc.
+
+You can load the Nix development with the command:
+```shell
+cd deku
+nix develop -c $SHELL
+```
+
+We recommend installing [`direnv`](https://direnv.net/) to automatically load the
+Nix environment for you whenever you enter the directory. Once installed, you can
+enable `direnv` for Deku with:
+```shell
+cd deku
+direnv allow
 ```
 
-(We suggest using this [binary cache repository](https://app.cachix.org/cache/anmonteiro) to reduce
-build times)
+### Building
+
+```
```

Comment:
Please add the "language" for the MD code sections (`shell script` here) everywhere in this file.

Reply:
done